### PR TITLE
test(js): Convert OrganizationApiKeys to RTL

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/views/settings/organizationApiKeys/index.spec.jsx [hunk 1]
```diff
@@ -54,14 +58,12 @@ describe('OrganizationApiKeys', function () {
...
     expect(deleteMock).toHaveBeenCalledTimes(0);
-    wrapper.find('Confirm[aria-label="Remove API Key"]').simulate('click');
+    userEvent.click(screen.getByRole('link', {name: 'Remove API Key?'}));
 
-    const modal = await mountGlobalModal();
-    modal.find('button[aria-label="Confirm"]').simulate('click');
-
-    wrapper.update();
+    renderGlobalModal();
+    userEvent.click(screen.getByTestId('confirm-button'));
```
Comment (priscilawebdev) [inline static/app/views/settings/organizationApiKeys/index.spec.jsx:64]: can we maybe look for `screen.getByRole('button', {name:'Confirm'})`?

Comment (evanpurkhiser) [reply static/app/views/settings/organizationApiKeys/index.spec.jsx:64]: I mentioned this somewhere else, but I think in the future I plan to go back through and kill the `confirm-button` data-test-id and replace all these with usages of `within`

Comment (priscilawebdev) [reply static/app/views/settings/organizationApiKeys/index.spec.jsx:64]: sorry, but why `within`? is there a case when we render multiple `Confirmation` buttons?

**Review by priscilawebdev — APPROVED** [review]

File: static/app/views/settings/organizationApiKeys/index.spec.jsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/views/settings/organizationApiKeys/index.spec.jsx:64]: Sometimes there are multiple buttons on the page with the same name.

Like i've seen a button for a row called like `Delete Item` and then in the confirm modal we name the confirm button, also, `Delete Item`. So when you try to click the confirm button, it picks-up BOTH buttons

**Pull request merged**